go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -290,11 +290,15 @@ func checkoutRemoteBranch(ctx *sql.Context, dSess *dsess.DoltSession, dbName str
 	}
 
 	if len(remoteRefs) == 0 {
+		if isTag, err := actions.IsTag(ctx, dbData.Ddb, branchName); err != nil {
+			return "", err
+		} else if isTag && apr.Contains(cli.MoveFlag) {
+			return "", fmt.Errorf(`dolt does not support a detached head state. To checkout a tag, run: 
+	dolt checkout %s -b {new_branch_namee}`, branchName)
```

Comment:
```suggestion
	dolt checkout %s -b {new_branch_name}`, branchName)
```

Reply:
Nice error message btw! Really, **really** helpful to tell users exactly what command they need to run whenever we can.